src/rule-unused-fragments.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict';
+
+const utils = require('./utils');
+
+const getGraphQLAST = utils.getGraphQLAST;
+
+function getGraphQLFragmentNames(graphQLAst) {
+  const fragmentNames = {};
+
+  function walkAST(node) {
+    if (node.kind === 'FragmentSpread') {
```

Comment:
We also check if the `FragmentSpread` node has any of the following directives. If it does, we skip adding it to your list of seen fragments.

1. `@module` - We skip this because this relates to an upcoming Relay feature (@josephsavona briefly mentions this at the [end of his ReactConf talk](https://www.youtube.com/watch?v=Tl0S7QkxFE4)). You can refer to this [example](https://astexplorer.net/#/gist/574016094d51f94e3f9ce4425198b4ac/40181766878bf82e2ac904ca92db5e9ff4cd2d11) for how it's used
2. `@relay(mask: false)` - This directive is [not recommended](https://relay.dev/docs/en/next/graphql-in-relay#relaymask-boolean), but we want to skip this because when this is false, the parent is explicitly reading the child fragment's data. Hence not skipping over this will introduce a false positive.